compiler/codegen.js

```diff
@@ -358,7 +358,12 @@ const generateIdent = (scope, decl) => {
 
     if (local?.idx === undefined) return internalThrow(scope, 'ReferenceError', `${unhackName(name)} is not defined`, true);
 
-    return [ [ Opcodes.local_get, local.idx ] ];
+    return [
+      [ Opcodes.local_get, local.idx ],
+      // todo: no support for i64
+      ...(valtypeBinary === Valtype.f64 && local.type === Valtype.i32 ? [ Opcodes.i32_from_u ] : []),
```

Comment:
These custom ops already do this sort of valtype check for us, you should be able to just check `local.type`?